### PR TITLE
fix: fix userdata resolve logic in v5 migration

### PR DESCRIPTION
**Pull request by frankqianms**
E2E TEST: N/A
To avoid end of line charactor resolving error, we change the logic here. The bad resolving logic may lead to bot-password and bot-id generating error in some projects when provision: https://msazure.visualstudio.com/Microsoft%20Teams%20Extensibility/_workitems/edit/17496886

### CODE REVIEW

**Comment by blackchoey** [comment]
please add some test cases for related bug to avoid regression

**Comment by frankqianms** [comment]
> please add some test cases for related bug to avoid regression



> please add some test cases for related bug to avoid regression

done.

File: packages/fx-core/tests/core/middleware/projectMigrationV3.test.ts [hunk 1]
```diff
@@ -1082,6 +1084,24 @@ describe("userdataMigration", () => {
...
+    sandbox
+      .stub(fs, "readFileSync")
+      .returns(
+        "fx-resource-aad-app-for-teams.clientSecret=abcd\nfx-resource-bot.botPassword=1234\n"
+      );
+
+    const migrationContext = await mockMigrationContext(projectPath);
+    await copyTestProject(Constants.happyPathTestProject, projectPath);
+    await userdataMigration(migrationContext);
+
```
Comment (blackchoey) [inline packages/fx-core/tests/core/middleware/projectMigrationV3.test.ts:1098]: can we call sandbox.restore() here? The current code works fine, my only concern is the mock may impact `assertFileContent()` function if someone changed its implementation to use `readFileSync()`. Restore the mocked function in advance can avoid such kind of issues.

Comment (frankqianms) [reply packages/fx-core/tests/core/middleware/projectMigrationV3.test.ts:1099]: reasonable concern. modified in the latest commit

**Review by blackchoey — APPROVED** [review]

**Review by xzf0587 — APPROVED** [review]

**Review by blackchoey — APPROVED** [review]

**Pull request merged**